### ISSUE TITLE
[Epic] Research MVP

### ISSUE THREAD

**Issue by PhoenixSss** [PhoenixSss/tracequant#1]
### Outcome

交付完整 P1 Research：从 Binance 公共 Raw 到可重现的规则、线性、LightGBM、XGBoost 比较和 Nautilus 事件验真，输出明确成本、时间可得性、失败实验与执行限制的研究报告。负结果是有效研究结果；软件完成不保证存在可晋升候选。

### Why

将数据正确性、因果验证、组合/风险/成本与实验身份连成可审计链路，为后续 Shadow、Demo 和受控 Live 提供可靠输入。早期规则结果用于反馈，不能代替完整 Research。

### Scope

设计来源：产品需求规格、总体设计、阶段交付计划 v1.0（2026-09-08）；本文固化适用边界。M01–M11 按领域职责协作，M12 提供最小研究入口，复用已完成 F0/E0。

- DP01–DP04：BTCUSDT/ETHUSDT 的 contract、mark、index 1m、settled funding，archive 与有界 REST recent/gap 到不变 Raw；USDC 可用性独立评估。
- DP05–DP07：canonical/DQ/PIT、15m/30m/1h/2h/4h、版本化因果 catalog、标签/样本、split 应用与查询。
- DP08–DP12：规则与对照、独立单标的 sizing/risk、统一成本和向量结果、线性/LGBM/XGB fold/OOF 比较。
- DP13–DP14：Nautilus 有限执行 policy/场景验真，协议/追踪/registry/报告与复现；最小实验身份先供 S02 消费。
- DP15.a：复用配置/日志的非交互研究入口；实时 runtime 在后续阶段履行。

### Non-goals

- 私有账户调用、Demo/Live 订单、生产凭据和真实资金授权。
- 将 USDC 评估当成 USDC 实盘证明；默认获取全部辅助衍生品或自动扩标的。
- 在线运行、真实账本、生产发布/HA、多交易所、高频、RL、自动放量。
- 收益保证、final-test 选参、无限特征/模型/搜索、为数量而建设平台。

### Success / Exit Criteria

- [ ] 所有本版 P1 产品义务及必要 Feature 依各自标准完成，基础能力复用而非重建。
- [ ] 四类数据在明确范围可获取/验证/恢复，缺失、重复、乱序、来源冲突可检测；USDC 结论真实且独立。
- [ ] 五种批准决策周期、完整首批 catalog、标签/样本/查询与无泄漏证据成立。
- [ ] 规则、long/flat 对照、线性/LGBM/XGB 在同一时序/成本协议比较；OOF 与最终测试纪律成立。
- [ ] signal/order/fill/return 时点清楚，sizing/risk 独立，毛净和 fee/funding/slippage/turnover 无重复计入。
- [ ] 向量/事件配对、maker/partial/cancel/latency 场景与差异/保真度限制可解释。
- [ ] 固定数据/代码/依赖/配置/种子与工件身份可重建；失败和负结果可查询。
- [ ] 形成完整研究阶段报告与候选晋升结论；没有合格候选时准确报告，不能伪造准入。

### Additional context

本次将旧 Epic 的“四个合约均须下载”明确为两个 USDT 研究标的与两个 USDC 的独立可用性评估；公开持仓量/其他辅助来源按明确消费者后置。将旧“一个基础 ML 过滤器”对齐为完整规则/线性/LGBM/XGB 与双层回测，并补齐原缺失的 30m/2h。子 Feature 中仍有效的数据、泄漏、成本、安全要求继续履行。

阶段安排为 S01 数据、S02 首个规则反馈、S03 完整 Research；需求/工作包在实施前有限冻结，Task 正文可以稍后定稿。已完成 #2 和 #11 子项合同/证据不改写，Epic/Feature 完成仍需既有审计，不能由计划正文宣告。

### Cross-cutting requirements

REQ-NF-01–12 按适用消费者持续履行：UTC/精度与区间；精确数值；可复现身份与有限容差；失败可见/有界重试；Secret 隔离；默认无真实写与 import 无副作用；跨模式共享语义/公共兼容；有限资源与时限；可重建部署/单写者；不变证据保留；可操作入口/报告；当前仓库验证与独立 Review。指标、预算和证据类型事先固定，缺失不能填零或默认放行。

REQ-GV-01–04：需求/有限成果先冻结，新增义务经明确变更；Task closeout、Feature 审计、阶段验收和资金授权分别成立；使用现有 LCK，不借设计新增工程平台。正式 parent/blocker/状态仍以 GitHub 原生事实为准。

**Comment by PhoenixSss** [comment]
## Research MVP 规划基线

本评论用于明确 Research MVP 的交付边界、Feature 骨架、依赖关系和完成定义。后续 Task 可以滚动细化，但不得在没有记录的情况下改变本 Epic 的核心目标。

### Epic 目标

建立一套可重复运行、可验证、无未来数据泄漏的量化研究系统，使研究人员能够针对 Binance 永续合约完成以下完整流程：

1. 获取和保存公开市场数据；
2. 将原始数据转换为统一、可验证的标准数据；
3. 构建特征、标签和研究数据集；
4. 运行包含现实成本假设的历史回测；
5. 研究多因子或机器学习基线策略；
6. 输出可复现的指标、实验结果和研究报告。

Research MVP 的目标是验证研究流程和策略研究能力，不包含实盘交易能力。

### 初始研究范围

当前 Research MVP 以以下范围作为规划基准：

- 交易场所：Binance 永续合约；
- 主要标的：BTC 和 ETH；
- 报价资产：USDT 和 USDC；
- 主要时间周期：15 分钟至 4 小时；
- 数据来源：公开市场数据，不要求私有账户权限；
- 策略方向：多因子和机器学习研究；
- 成本假设：需要考虑手续费、maker/taker 差异、资金费率以及可解释的滑点假设；
- 结果用途：离线研究和历史验证。

范围可以通过后续 Issue 决策进一步收窄，但扩大范围必须明确记录理由。

### Feature 骨架

Research MVP 当前规划为以下 Feature：

1. `[Feature] 工程与研究基础`
   - Python 工程、依赖、测试、CI、配置、日志、时间规范和核心领域基础。

2. `[Feature] 市场数据采集与原始存储`
   - 获取公开市场数据，保存不可变原始数据，并支持可重复增量同步。

3. `[Feature] 标准化行情与数据质量`
   - 将原始数据转换为统一数据模型，执行完整性、唯一性、时间连续性和异常值检查。

4. `[Feature] 特征、标签与研究数据集`
   - 构建无未来数据泄漏的特征和标签，生成带有时间语义和版本信息的研究数据集。

5. `[Feature] 回测与绩效评估`
   - 建立历史回测流程，处理仓位、手续费、资金费率、滑点和绩效指标。

6. `[Feature] 基线策略与模型研究`
   - 实现可解释的基线策略、多因子基线和机器学习基线，并进行时间序列验证。

7. `[Feature] 实验复现与研究报告`
   - 保存实验输入、参数、代码版本、指标和产物，生成可审计的研究结论。

Feature 名称在创建时可以做轻微措辞调整，但职责边界应保持一致。

### 主要依赖关系

总体依赖方向为：

`工程基础 → 数据采集 → 数据标准化 → 研究数据集 → 回测 → 策略研究 → 研究报告`

具体关系：

- 工程与研究基础是其他所有 Feature 的共同依赖；
- 标准化行情依赖市场数据采集；
- 特征、标签和研究数据集依赖标准化行情；
- 回测依赖标准化行情，部分研究流程也依赖研究数据集；
- 基线策略与模型研究依赖研究数据集和回测；
- 实验复现与研究报告依赖回测和策略研究结果。

允许在接口清晰时并行推进，但不得绕过尚未定义的数据契约。

### Epic 完成定义

只有以下条件全部满足时，Research MVP 才能关闭：

- 所有规划 Feature 已完成或通过明确决策移出本 Epic；
- 工程可在全新环境中依据锁文件完成安装；
- pytest、Ruff 和 mypy 在 CI 中持续通过；
- 可以针对至少一个明确历史区间获取 BTC 或 ETH 永续合约数据；
- 原始数据和标准化数据具有清晰、可验证的时间语义；
- 数据质量检查能够发现重复、缺失、乱序和非法时间数据；
- 特征和标签计算具有防未来数据泄漏约束；
- 可以运行至少一个包含手续费和现实成本假设的历史回测；
- 至少存在一个可解释基线策略；
- 至少存在一个多因子或机器学习研究基线；
- 研究结果包含收益、风险、回撤、换手和成本相关指标；
- 实验能够通过固定的数据、参数、代码版本和随机种子复现；
- 存在一份清晰说明结论、限制和下一阶段建议的研究报告；
- 不需要 Binance 私有 API Key；
- 不包含 Shadow 或 Live 订单执行。

### 明确范围外

Research MVP 不包含：

- 实盘下单；
- Shadow 订单执行；
- 私有账户资产或持仓同步；
- API Key、Secret 或交易权限管理；
- 自动部署交易服务；
- 高可用交易基础设施；
- 低延迟或高频交易；
- 完整投资组合资金管理；
- 多交易所执行；
- 移动端或 Web 交易界面；
- Live MVP 的运维与告警能力。

### 滚动规划规则

- Epic 提前维护完整 Feature 骨架；
- Feature 在开始实施前必须明确目标、输入、输出、边界和完成定义；
- Task 只对近期工作进行详细拆分；
- 当前 Task 必须具有完整验收标准后才能标记为 `codex:ready`；
- 接下来 1～2 个 Task 可以处于 `Ready`；
- 同一时间原则上只有一个主要 Task 处于 `In Progress`；
- 更远的任务先记录在 Feature 规划中，不急于创建 Issue；
- 每完成 2～4 个 Task，重新审查一次所属 Feature 的剩余范围；
- 新增、删除或重排 Feature 时，必须在 Epic 中留下规划变更记录。

**Comment by PhoenixSss** [comment]
## Research MVP Feature 骨架修订 v2

本评论修订此前规划基线中的 Feature 骨架。

此前已经确定的 Epic 目标、初始研究范围、范围外内容、滚动规划规则和总体完成定义继续有效。本次修订只调整 Feature 划分、职责边界和依赖关系。

修订原因：

1. 前期技术研究已经确认采用“向量化研究粗筛 + NautilusTrader 事件驱动回测验真”的双层验证路线；
2. 向量化回测和事件驱动回测具有不同目标、输入、复杂度和验收标准，不应合并为一个过大的 Feature；
3. 数据采集需要明确限定为 Binance 公共市场数据和不可变原始数据湖；
4. 实验复现不仅包含研究报告，还需要覆盖数据版本、模型登记、成本配置和代码版本；
5. Research MVP 必须严格隔离私有账户、Demo 和 Live 能力。

本评论中的 Feature 骨架取代此前评论中的 Feature 列表，但不追溯删除旧评论。

---

## 修订后的 Feature 骨架

### 1. `[Feature] 工程与研究基础`

现有 Issue：

- #2

职责：

- Python 工程和依赖管理；
- 测试、lint、格式化和类型检查；
- GitHub Actions CI；
- UTC 时间规范；
- 类型化配置；
- 结构化日志；
- 核心基础类型；
- 路径与数据目录规范；
- 开发与架构文档。

该 Feature 不包含行情数据、特征、回测、策略或模型逻辑。

---

### 2. `[Feature] Binance 公共市场数据与原始数据湖`

职责：

- Binance USDⓈ-M 公共历史市场数据；
- 官方历史文件回填；
- REST 最近数据同步和缺口补采；
- BTCUSDT、ETHUSDT 首批支持；
- USDC 合约数据可用性确认；
- 1m 基础历史数据；
- 标记价格、指数价格、已结算资金费率等公开衍生品数据；
- 不可变 Raw Parquet；
- manifest、校验、原子写入、幂等和恢复。

该 Feature 不执行标准化、周期聚合、缺失填补、特征计算或账户数据访问。

---

### 3. `[Feature] 标准化行情与数据质量`

职责：

- 自研 canonical schema；
- 统一 symbol、market、source 和时间语义；
- Raw 数据解析和标准化；
- 1m 数据质量检查；
- 重复、缺失、乱序、非法值和时间连续性检查；
- 从统一 1m 基础数据聚合 15m、30m、1h、2h 和 4h；
- 数据版本和质量报告；
- Polars 变换和 DuckDB 查询入口。

该 Feature 不采集原始数据，也不计算策略特征或标签。

---

### 4. `[Feature] 特征、标签与研究数据集`

职责：

- 可解释基础特征；
- 趋势、反转、波动率、成交量、基差、资金费率和持仓结构特征；
- 标签和净收益目标；
- fee、funding 和滑点相关标签语义；
- point-in-time 可得性；
- lookback、warm-up 和缺失值规则；
- 防未来数据泄漏约束；
- 可版本化研究数据集；
- 训练、验证和测试输入契约。

该 Feature 不训练正式模型，不执行事件驱动订单仿真。

---

### 5. `[Feature] 向量化研究回测与绩效评估`

职责：

- 轻量自研向量化回测；
- 因子和规则策略快速筛选；
- 仓位和信号基本语义；
- 手续费、资金费率和粗滑点；
- 收益、回撤、换手和风险指标；
- 成本敏感性；
- 参数扰动；
- walk-forward、purging 和 embargo；
- 样本外评估；
- 候选策略筛选报告。

该 Feature 负责快速研究筛选，不作为最终执行真实性验收。

---

### 6. `[Feature] NautilusTrader 事件驱动回测与成本仿真`

职责：

- NautilusTrader 回测环境；
- 标准数据到 NautilusTrader 输入的适配边界；
- 订单生命周期；
- maker/taker 路径；
- post-only / GTX 语义；
- 部分成交、撤单、超时和重新报价；
- fee、funding、fill 和 latency model；
- maker-first 执行 policy 的历史仿真；
- 与向量化回测结果的差异分析；
- 候选策略执行可行性验收。

该 Feature 不接入 Binance 私有 API，不运行 Demo 或 Live Node。

---

### 7. `[Feature] 基线策略与机器学习研究`

职责：

- 可解释规则基线；
- 简单统计或线性基线；
- LightGBM 主模型；
- XGBoost Challenger；
- 特征选择和重要性分析；
- 时间序列训练与样本外预测；
- 状态切换和多因子组合；
- 模型与规则策略对照；
- 候选策略进入事件回测的准入条件。

深度学习、GNN、强化学习和 LLM 直接交易不属于本 Feature 的首期范围。

---

### 8. `[Feature] 实验复现、模型登记与研究报告`

职责：

- 实验输入和配置记录；
- 原始数据、标准化数据、特征和标签版本；
- Git commit 和依赖环境；
- 随机种子；
- 模型参数和产物；
- 回测引擎与成本配置；
- MLflow 实验追踪和模型登记；
- 指标、图表和结果摘要；
- 已知限制和失败实验记录；
- Research MVP 最终研究报告；
- 是否进入 Shadow & Demo MVP 的建议。

该 Feature 不负责模型在线发布、实盘审批或生产回滚。

---

## 主要依赖关系

总体依赖方向：

```text
工程与研究基础
        ↓
Binance 公共市场数据与原始数据湖
        ↓
标准化行情与数据质量
        ↓
特征、标签与研究数据集
        ↓
向量化研究回测与绩效评估
        ↓
基线策略与机器学习研究
        ↓
NautilusTrader 事件驱动回测与成本仿真
        ↓
实验复现、模型登记与研究报告
```

更精确的依赖说明：

- 所有 Feature 依赖工程与研究基础；
- 原始数据湖是标准化行情的上游；
- 特征和标签依赖标准化数据；
- 向量化回测依赖标准化行情以及特征、标签；
- 基线策略和机器学习依赖研究数据集和向量化回测；
- NautilusTrader 事件回测依赖标准化行情、候选策略和成本假设；
- 实验复现与研究报告依赖其他 Research Feature 的可追溯产物。

Feature 可以在接口明确后部分并行，但不得绕过尚未冻结的数据契约。

---

## Research MVP 阶段边界

本 Epic 只包含：

```text
Binance 公共数据
→ 不可变 Raw Parquet
→ Canonical schema 与数据质量
→ Polars / DuckDB 研究层
→ 特征与标签
→ 向量化研究筛选
→ LightGBM / XGBoost 与规则基线
→ NautilusTrader 事件驱动验真
→ 可复现研究报告
```

本 Epic 明确不包含：

- Binance 私有 API；
- API Key 或 Secret；
- user data stream；
- 账户余额、订单、成交和持仓；
- Binance Demo 下单；
- Shadow 实时运行；
- Live Node；
- 真实资金；
- PostgreSQL 实盘真相账本；
- 生产告警和值班；
- L2/L3 付费历史数据；
- 多交易所生产适配；
- Kubernetes。

这些能力分别属于后续 Shadow & Demo MVP、Live MVP 或 Production Hardening & Expansion。

---

## Research MVP 关闭门禁

只有以下条件全部满足时，本 Epic 才能关闭：

- 8 个规划 Feature 全部完成，或通过明确决策移出本 Epic；
- 工程可以在全新环境中依据锁文件安装；
- pytest、Ruff、mypy 和 CI 持续通过；
- 至少可以重建一个明确历史区间的 BTC 或 ETH 1m 公共数据；
- Raw 数据、标准化数据和研究数据集具有明确版本；
- 重复、缺失、乱序和非法数据可被检测；
- 15m～4h 数据由统一 1m 数据聚合；
- 特征和标签通过未来数据泄漏检查；
- 至少存在一个可解释规则基线；
- 至少存在一个 LightGBM 或 XGBoost 模型基线；
- 向量化回测包含 fee、funding 和滑点；
- 候选策略通过 NautilusTrader 事件驱动回测；
- 研究结果包含收益、风险、回撤、换手和成本指标；
- 数据、参数、随机种子、代码版本和产物可追溯；
- 存在明确说明结论、限制和后续建议的研究报告；
- 未引入任何私有账户或真实下单能力。

---

## 当前规划状态

- 当前只修订 Feature 骨架；
- 暂不一次性创建全部 Task；
- Feature 创建后统一保持 `Specifying` 和 `codex:needs-spec`；
- 仅在某个 Feature 即将实施时滚动细化近期 Task；
- 同一时间原则上只有一个主要 Task 处于 `In Progress`。
